cpp/src/parquet/encoding.cc

```diff
@@ -1486,8 +1486,9 @@ class DictDecoderImpl : public DecoderImpl, virtual public DictDecoder<Type> {
       return;
     }
     uint8_t bit_width = *data;
-    if (ARROW_PREDICT_FALSE(bit_width >= 64)) {
-      throw ParquetException("Invalid or corrupted bit_width");
+    if (ARROW_PREDICT_FALSE(bit_width > 32)) {
+      throw ParquetException("Invalid or corrupted bit_width " +
+                             std::to_string(bit_width) + ". Maximum allowed is 32.");
```

Comment:
Speaking of which, is 0 accepted as a bit width or should be raise an error in that case also?

Reply:
0 is okay as far as I can tell. I created a dictionary encoded byte array column that was all nulls. The resulting bit width was 0 and was readable by both parquet cpp and parquet mr.